src/main/java/org/breedinginsight/brapps/importer/model/imports/germplasm/GermplasmImportService.java

```diff
@@ -0,0 +1,263 @@
+import tech.tablesaw.api.Table;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Singleton
+public class GermplasmImportService extends BrAPIImportService {
+
+    private String IMPORT_TYPE_ID = "GermplasmImport";
+
+    private BrAPIGermplasmDAO brAPIGermplasmDAO;
+    private BrAPIProgramDAO brAPIProgramDAO;
+
+    @Inject
+    public GermplasmImportService(FileMappingUtil fileMappingUtil,
+                                  BrAPIProgramDAO brAPIProgramDAO, BrAPIGermplasmDAO brAPIGermplasmDAO, BrAPICrossDAO brAPICrossDAO)
+    {
+        this.brAPIGermplasmDAO = brAPIGermplasmDAO;
+        this.brAPIProgramDAO = brAPIProgramDAO;
+    }
+
+    @Override
+    public GermplasmImport getImportClass() {
+        return new GermplasmImport();
+    }
+
+    @Override
+    public String getImportTypeId() {
+        return IMPORT_TYPE_ID;
+    }
+
+    @Override
+    public ImportPreviewResponse process(List<BrAPIImport> brAPIImports, Table data, Program program, Boolean commit) throws UnprocessableEntityException {
+
+        //BrAPI Objects per row
+        List<GermplasmImport> germplasmImports = (List<GermplasmImport>)(List<?>) brAPIImports;
+
+        // Get BrAPI Program
+        BrAPIProgram brAPIProgram;
+        try {
+            Optional<BrAPIProgram> optionalBrAPIProgram = brAPIProgramDAO.getProgram(program.getId());
+            if (optionalBrAPIProgram.isEmpty()) throw new ApiException("Program was not found in the brapi service");
+            brAPIProgram = optionalBrAPIProgram.get();
+        } catch (ApiException e) {
+            // Our program should be set up already
+            throw new InternalServerException(e.toString(), e);
+        }
+
+        // Get all of our objects specified in the data file by their unique attributes
+        Set<String> germplasmNames = new HashSet<>();
+        for (int i = 0; i < germplasmImports.size(); i++) {
+            GermplasmImport germplasmImport = germplasmImports.get(i);
+            if (germplasmImport.getGermplasm() != null){
+                if (germplasmImport.getGermplasm().getGermplasmName() != null) {
+                    germplasmNames.add(germplasmImport.getGermplasm().getGermplasmName());
+                }
+
+                if (germplasmImport.getGermplasm().getFemaleParent() != null) {
+                    germplasmNames.add(germplasmImport.getGermplasm().getFemaleParent().getReferenceValue());
+                }
+
+                if (germplasmImport.getGermplasm().getMaleParent() != null) {
+                    germplasmNames.add(germplasmImport.getGermplasm().getMaleParent().getReferenceValue());
+                }
+            }
+        }
+
+        // Setting up our data elements
+        List<GermplasmImportPending> mappedBrAPIImport = germplasmImports.stream()
+                .map(germplasmImport -> new GermplasmImportPending()).collect(Collectors.toList());
```

Comment:
Why do you create all of these objects ahead of time?  Couldn't you create them during processing below and add them to this list?